paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/format/RecordParser.java

```diff
@@ -141,10 +143,27 @@ public void flatMap(String value, Collector<RichCdcMultiplexRecord> out) {
     protected Map<String, String> extractRowData(
             JsonNode record, LinkedHashMap<String, DataType> paimonFieldTypes) {
         paimonFieldTypes.putAll(fillDefaultStringTypes(record));
-        Map<String, String> recordMap =
-                OBJECT_MAPPER.convertValue(record, new TypeReference<Map<String, String>>() {});
-
-        Map<String, String> rowData = new HashMap<>(recordMap);
+        Map<String, Object> recordMap =
+                OBJECT_MAPPER.convertValue(record, new TypeReference<Map<String, Object>>() {});
+        Map<String, String> rowData =
+                recordMap.entrySet().stream()
+                        .collect(
+                                Collectors.toMap(
+                                        Map.Entry::getKey,
+                                        entry -> {
+                                            if (Objects.nonNull(entry.getValue())
+                                                    && !TypeUtils.isBasicType(entry.getValue())) {
+                                                try {
+                                                    return OBJECT_MAPPER
+                                                            .writer()
+                                                            .writeValueAsString(entry.getValue());
+                                                } catch (JsonProcessingException e) {
+                                                    LOG.error("Failed to deserialize record.", e);
+                                                    return Objects.toString(entry.getValue());
+                                                }
+                                            }
+                                            return Objects.toString(entry.getValue(), null);
+                                        }));
```

Comment:
How about remove `!TypeUtils.isBasicType(entry.getValue())` check ?

Reply:
Because the string changes from "a" -> ""a"" after using the json writeValueAsString method.